source/server/debug_session_properties_restricted_service.cpp

```diff
@@ -0,0 +1,176 @@
+  SetDebugSessionEnabledResponse* response)
+{
+  uint32_t enabled = request->enabled() ? 1 : 0;
+  return set_uint_property(context, (NISysCfgResourceProperty)0x10002000, request->device_id(), enabled);
+}
+
+::grpc::Status DebugSessionPropertiesRestrictedService::SetDebugSessionServerOutOfProcess(
+  ::grpc::ServerContext* context,
+  const SetDebugSessionServerOutOfProcessRequest* request,
+  SetDebugSessionServerOutOfProcessResponse* response)
+{
+  return set_bool_property(context, (NISysCfgResourceProperty)0x10003000, request->device_id(), request->out_of_process());
+}
+
+::grpc::Status DebugSessionPropertiesRestrictedService::access_syscfg_resource_by_device_id_filter(
+  ::grpc::ServerContext* context,
+  const nidevice_restricted_grpc::DeviceId& device_id,
+  std::function<NISysCfgStatus(nidevice_grpc::SysCfgLibraryInterface*, NISysCfgResourceHandle, bool*)> syscfg_resource_action_func)
+{
+  NISysCfgStatus status = NISysCfg_OK;
+  NISysCfgSessionHandle session = NULL;
+  NISysCfgFilterHandle filter = NULL;
+  NISysCfgEnumResourceHandle resources_handle = NULL;
+  NISysCfgResourceHandle resource = NULL;
+
+  try {
+    auto library = get_syscfg_library_interface();
+    if (NISysCfg_Succeeded(status = open_or_get_localhost_syscfg_session(&session))) {
+      if (NISysCfg_Succeeded(status = library->CreateFilter(session, &filter))) {
+        library->SetFilterProperty(filter, NISysCfgFilterPropertyIsDevice, NISysCfgBoolTrue);
+        library->SetFilterProperty(filter, NISysCfgFilterPropertyUserAlias, device_id.name().c_str());
+        library->SetFilterProperty(filter, NISysCfgFilterPropertySerialNumber, device_id.serial_number().c_str());
+        library->SetFilterProperty(filter, NISysCfgFilterPropertyProductId, device_id.product_id());
+        if (NISysCfg_Succeeded(status = library->FindHardware(session, NISysCfgFilterModeMatchValuesAll, filter, NULL, &resources_handle))) {
+          if (NISysCfg_Succeeded(status) && (status = library->NextResource(session, resources_handle, &resource)) == NISysCfg_OK) {
+            bool save_changes = false;
+            if (NISysCfg_Succeeded(status = syscfg_resource_action_func(library, resource, &save_changes))) {
+              if (save_changes) {
+                NISysCfgBool changes_require_restart = NISysCfgBoolFalse;
+                char* detailed_changes = nullptr;
+                status = library->SaveResourceChanges(resource, &changes_require_restart, &detailed_changes);
+                library->FreeDetailedString(detailed_changes);
+              }
+            }
+            library->CloseHandle(resource);
+          }
+          library->CloseHandle(resources_handle);
+        }
+        library->CloseHandle(filter);
+      }
+    }
+  }
+  catch (nidevice_grpc::NonDriverException& ex) {
+    return ex.GetStatus();
+  }
+
+  if (NISysCfg_Failed(status)) {
+    return ::grpc::Status(::grpc::StatusCode::INTERNAL, kDebugSessionPropertyAccessFailedMessage);
```

Comment:
The others use status code `UNKNOWN`. Use [ApiErrorAndDescriptionToStatus](https://github.com/ni/grpc-device/blob/main/source/server/converters.h#L325) instead of creating this yourself.

Reply:
Updated as suggested while leaving the static message in place.